core/src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -619,24 +620,22 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     case class SampleConf(a: Int, b: String)
     loadConfig[SampleConf](conf).failure.exception shouldEqual KeyNotFoundException("a")
 
-    implicit val mapping = new ConfigFieldMapping[SampleConf] {
-      def apply(fieldName: String) = fieldName.toUpperCase
-    }
+    implicit val productHint = ProductHint[SampleConf](ConfigFieldMapping(_.toUpperCase))
 
     loadConfig[SampleConf](conf) shouldBe Success(SampleConf(2, "two"))
   }
 
   it should "allow customizing the field mapping with word delimiters" in {
```

Comment:
Suggestion: Update this test's name to indicate that it is testing CamelCase instead of word delimiters.